web-app/src/main/java/com/google/sps/data/QuadTree.java

```diff
@@ -56,4 +58,85 @@ public void printTree() {
     System.out.printf("%n");
   }
 
+  public void insert(PoliceReport report) {
+    Node leafNode = findLeaf(root, report);
+    leafNode.numReports += 1;
+    leafNode.reports.add(report);
+    // If max capacity has been exceeded, create child nodes
+    if (leafNode.numReports > reportCapacity && leafNode.depth < maxDepth) {
+      leafNode.children = reallocateReports(leafNode.reports, leafNode.coordinates, leafNode.depth);
+      leafNode.leaf = false;
+      leafNode.reports = null;
+    }
+  }
+
+  // Find a leaf to insert the report in
```

Comment:
I created a new method for findLeaf, but I'm not too sure what it adds to my code. In the future should I aim to split up my functions wherever possible?

Reply:
Usually, a sign that your method is doing multiple things is if you need to have comments to delineate the different sections of your method.